test/tests/serial/open_dkim.bats

```diff
@@ -1,465 +1,288 @@
-  assert_output 4
+function __assert_has_entry_in_signingtable() {
+  __assert_output_for 'SigningTable' "${@}"
 }
 
-@test "${TEST_FILE}generator creates keys, tables and TrustedHosts using manual provided domain name" {
-  local PRIVATE_CONFIG
-  PRIVATE_CONFIG=$(duplicate_config_for_container . "${BATS_TEST_NAME}")
-  rm -rf "${PRIVATE_CONFIG}/with-domain" && mkdir -p "${PRIVATE_CONFIG}/with-domain"
-
-  # generate first key
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/":/tmp/docker-mailserver/ \
-    "${IMAGE_NAME}" /bin/bash -c 'open-dkim keysize 2048 domain domain1.tld | wc -l'
-
-  assert_success
-  assert_output 4
-
-  # generate two additional keys different to the previous one
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/":/tmp/docker-mailserver/ \
-    "${IMAGE_NAME}" /bin/bash -c 'open-dkim keysize 2048 domain "domain2.tld,domain3.tld" | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # generate an additional key whilst providing already existing domains
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/":/tmp/docker-mailserver/ \
-    "${IMAGE_NAME}" /bin/bash -c 'open-dkim keysize 2048 domain "domain3.tld,domain4.tld" | wc -l'
-
-  assert_success
-  assert_output 1
-
-  # check keys for domain1.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain1.tld/ | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check keys for domain2.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain2.tld | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check keys for domain3.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain3.tld | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check keys for domain4.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain4.tld | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check presence of tables and TrustedHosts
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c "ls -1 /etc/opendkim | grep -E 'KeyTable|SigningTable|TrustedHosts|keys' | wc -l"
+function __assert_logged_dkim_creation() {
+  __assert_output_for 'Log' "${@}"
+}
 
-  assert_success
-  assert_output 4
+function __assert_output_for() {
+  local ASSERT_FOR=${1}
+  local EXPECTED_DOMAIN=${2}
+  local EXPECTED_SELECTOR=${3:-'mail'}
+
+  case "${ASSERT_FOR}" in
```

Comment:
Do we want / need a default `( * )` for this switch-case statement? Usually good practice to catch issues.

Reply:
I can add one, but it's only a method used by the direct 3 wrapping methods preceding it. And only this test would use this method.

Reply:
Should I just remove the switch statement method, and copy/paste the 3 parameters into the wrapper ones with their relevant assertion? :thinking:

Reply:
> Should I just remove the switch statement method, and copy/paste the 3 parameters into the wrapper ones with their relevant assertion? 🤔

This may actually be the best idea, yes. It removes the need for the default case and eliminates some complexity as well.